base/applications/rapps/appview.cpp

```diff
@@ -970,16 +973,89 @@ CAppInfoDisplay::~CAppInfoDisplay()
 
 // **** CAppsListView ****
 
+struct CAsyncLoadIcon {
+    CAsyncLoadIcon *pNext;
+    HWND hAppsList;
+    CAppInfo *AppInfo; // Only used to find the item in the list, do not access on background thread
+    UINT TaskId;
+    bool Parse;
+    WCHAR Location[1];
+
+    void Free() { LocalFree(this); }
+    static CAsyncLoadIcon* Queue(HWND hAppsList, CAppInfo &AppInfo, bool Parse);
+    static void StartTasks();
+} *g_AsyncIconTasks = NULL;
+static UINT g_AsyncIconTaskId = 0;
+
+static DWORD CALLBACK
+AsyncLoadIconProc(LPVOID Param)
+{
+    for (CAsyncLoadIcon *task = (CAsyncLoadIcon*)Param, *old; task; old->Free())
```

Comment:
What's this "old" variable ? Where is it defined?

Reply:
What do you mean, it's defined right there. It is walking the linked list and you can't access the next pointer after freeing.